# Project 2
Analyzing data from csv datasets found on the web using Pandas

For this project I am going to analyze three different datasets using Pandas.  

The first dataset is an Airbnb dataset found at, first I'll load up the dataset into a pandas dataframe

In [43]:
import pandas as pd
import numpy as np

airbnb_data=pd.read_csv('airbnb_listings.csv')
airbnb_data

id                                               name    host_id  \
0          2454                                         superCondo       2688   
1          2539                 Clean & quiet apt home by the park       2787   
2          2595                              Skylit Midtown Castle       2845   
3          3330                 ++ Brooklyn Penthouse Guestroom ++       4177   
4          3647                THE VILLAGE OF HARLEM....NEW YORK !       4632   
5          3831                    Cozy Entire Floor of Brownstone       4869   
6          5022   Entire Apt: Spacious Studio/Loft by central park       7192   
7          5099          Large Cozy 1 BR Apartment In Midtown East       7322   
8          5121                                    BlissArtsSpace!       7356   
9          5178                   Large Furnished Room Near B'way        8967   
10         5203                 Cozy Clean Guest Room - Family Apt       7490   
11         5216                                    Super Large APT       7510   
12         5238                 Cute & Cozy Lower East Side 1 bdrm       7549   
13         5295                   Beautiful 1br on Upper West Side       7702   
14         5441                    Central Manhattan/near Broadway       7989   
15         5586                 Rooftop Deck/City Views. Great Apt       8526   
16         5803     Lovely Room 1, Garden, Best Area, Legal rental       9744   
17         6021   Wonderful Guest Bedroom in Manhattan for SINGLES      11528   
18         6090                      West Village Nest - Superhost      11975   
19         6848                   Only 2 stops to Manhattan studio      15991   
20         6990                 UES Beautiful Blue Room / Oak Room      16800   
21         7036                 Large Guest Room for Broolyn Condo      53179   
22         7097                  Perfect for Your Parents + Garden      17571   
23         7322                                    Chelsea Perfect      18946   
24         7726    Hip Historic Brownstone Apartment with Backyard      20950   
25         7801                   Sweet and Spacious Brooklyn Loft      21207   
26         7970                                       Chic Chelsea      10496   
27         8024                     CBG CtyBGd HelpsHaiti rm#1:1-4      22486   
28         8110                              CBG Helps Haiti Rm #2      22486   
29         8343          Quiet, Spacious 3BR, 1.5 BA EV apt w/yard      24222   
...         ...                                                ...        ...   
50198  32045241    Beautiful 3Bedroom Apt in the Upper East Side!!   38082447   
50199  32045303        Lux one bedroom available in trendy Nolita!  236839319   
50200  32045403          Comfortable Private Room in Crown Heights   64213111   
50201  32045441          Beautiful 3 bed 1 bath in Iconic Bed-Stuy    1405865   
50202  32045763                             Home away by the beach  234781729   
50203  32046149                             Cozy home by the beach  234781729   
50204  32047002  3 JFK Layover - Express train to Manhattan-30 ...   67746251   
50205  32047800                  Cozy Room starting at $67 a night    9898029   
50206  32047965   Private bdrm, living room, bathroom, kitchenette  240283485   
50207  32048480                       Calm, Quiet Space in Bedstuy   60403058   
50208  32048560        Cosy one bedroom apartment in Hells Kitchen  240289379   
50209  32048862        Brooklyn's Cutest Twin Beds W/Spa Amenities   89985620   
50210  32048883                Bayridge Brooklyn Beauty Townhouse!  240291101   
50211  32048953                          Charming Meatpacking Loft    5083078   
50212  32050059                          Comfortable & Simple Room     592967   
50213  32050071       Modern luxury apartment in the heart of Noho    6073347   
50214  32050188         Private room in the heart of Manhattan NYC  239519185   
50215  32050215                Stylish Apart

# First question - what is the average price of a private room in Manhattan?
Let's dig into the pandas dataframe to find apartments in Manhattan, then, Private rooms, and finally run an average of the price

In [5]:
ma_data = airbnb_data[airbnb_data['neighbourhood_group'].str.contains("Manhattan")]
ma_pr_data = ma_data[ma_data['room_type'].str.contains("Private room")]
ma_pr_data

id                                               name    host_id  \
4          3647                THE VILLAGE OF HARLEM....NEW YORK !       4632   
9          5178                   Large Furnished Room Near B'way        8967   
10         5203                 Cozy Clean Guest Room - Family Apt       7490   
14         5441                    Central Manhattan/near Broadway       7989   
17         6021   Wonderful Guest Bedroom in Manhattan for SINGLES      11528   
20         6990                 UES Beautiful Blue Room / Oak Room      16800   
23         7322                                    Chelsea Perfect      18946   
32         8700  Magnifique Suite au N de Manhattan - vue Cloitres      26394   
34         9518       SPACIOUS, LOVELY FURNISHED MANHATTAN BEDROOM      31374   
36         9668                              front room/double bed      32294   
37         9704                Spacious 1 bedroom in luxe building      32045   
39         9783                                back room/bunk beds      32294   
46        12192                                ENJOY Downtown NYC!      46978   
49        12318                                  West Side Retreat      16800   
63        14468  Under the Empire State Building/Midtown Manhattan      56909   
66        14991                 Great Location, Manhattan Bedroom!      59023   
67        15220                Best Location in NYC! TIMES SQUARE!      59734   
79        16421                       Your Heaven in Hells Kitchen      63924   
81        16580                Sunny, Modern room in East Village!      64442   
83        16821             Large Room in Amazing East Village Apt       4396   
86        17092                        Cool Room in Hell's Kitchen      66243   
91        17693                    HARLEM, NEW YORK WELCOMES YOU!!      68428   
94        18152                                     Manhattan Room      69942   
98        18728                Beautiful Meatpacking District Loft      71876   
104       19319                      Private room Long Term Rental      44263   
107       20299                                Oh glorious spring!      62407   
108       20300                             Great Location for NYC      76627   
120       21644                          Upper Manhattan, New York      82685   
121       21794                        COZY QUIET room 4 DOOGLERS!      83257   
123       22918   loft bed - near transportation-15min to times sq      32294   
...         ...                                                ...        ...   
50097  32005497  NEW! Amazing room in 3 BR flat w/ rooftop in SOHO   17236151   
50101  32008158                Cozy dreamy located private getaway   94173816   
50108  32009894  Room 3 in Furnished Apartment- Beautiful 3 Bed...  239355304   
50128  32015007                          Cozy room next to subway!  240033083   
50134  32016041                    Spectacular Chinatown Apartment   14249909   
50135  32016160   Room in Fully Furnished - Spacious 3 Bedroom Apt  239355304   
50136  32016609                        Time square/ Hell’s Kitchen   98091426   
50138  32016869   1 Bed/shared bath in the heart of Nolita! (Soho)   24077660   
50146  32020853    ★ Clean, Private BR in Little Italy/Chinatown ★   14249909   
50149  32021345  Beautiful Room Amazing View w/Balcony 150 reviews  110020623   
50150  32021582  Private BIG BEDROOM with own BATHROOM 150(Revi...  110020623   
50151  32022587                       Peaceful modern room in SoHo    1475812   
50155  32023208                                   Comfortable home  233557783   
50156  32023903  Luxury Master Bedroom at Korea Town/Empire St ...  240108110   
50157  32025639                 Enjoy the View in a Cozy Lux Condo   56656728   
50158  32026654  AMAZING PRIVATE  BEDROOM BEST LOCATION 150 REV...  110020623   
50165  32036822  Private comfy and large room in Washington Hei...   36323224   
50166  32036850                      Charmin

In [6]:
ma_pr_data["price"].mean()

112.52065096782148

# The average price of a private room in Manhattan on airbnb is 113 a night

The second dataset is a set of rent data I found on streeteasy.net.

In [8]:
rent_data = pd.read_csv('medianAskingRent_All.csv')
rent_data

areaName        Borough      areaType  2010-01  2010-02  \
0           All Downtown      Manhattan     submarket   3200.0   3200.0   
1            All Midtown      Manhattan     submarket   2895.0   2800.0   
2    All Upper East Side      Manhattan     submarket   2469.0   2450.0   
3    All Upper Manhattan      Manhattan     submarket   1825.0   1800.0   
4    All Upper West Side      Manhattan     submarket   2895.0   2800.0   
5             Alley Park         Queens  neighborhood      NaN      NaN   
6                Astoria         Queens  neighborhood   1600.0   1650.0   
7             Auburndale         Queens  neighborhood      NaN      NaN   
8             Bath Beach       Brooklyn  neighborhood      NaN      NaN   
9      Battery Park City      Manhattan  neighborhood   3495.0   3346.0   
10             Bay Ridge       Brooklyn  neighborhood   1350.0   1388.0   
11            Baychester          Bronx  neighborhood      NaN      NaN   
12               Bayside         Queens  neighborhood   1775.0   1838.0   
13          Bedford Park          Bronx  neighborhood      NaN      NaN   
14    Bedford-Stuyvesant       Brooklyn  neighborhood   1650.0   1650.0   
15             Bellerose         Queens  neighborhood      NaN      NaN   
16               Belmont          Bronx  neighborhood      NaN      NaN   
17           Bensonhurst       Brooklyn  neighborhood      NaN      NaN   
18          Bergen Beach       Brooklyn  neighborhood      NaN      NaN   
19           Boerum Hill       Brooklyn  neighborhood   2600.0   2700.0   
20          Borough Park       Brooklyn  neighborhood      NaN      NaN   
21            Bowne Park         Queens  neighborhood      NaN      NaN   
22             Briarwood         Queens  neighborhood      NaN   1325.0   
23        Brighton Beach       Brooklyn  neighborhood      NaN      NaN   
24                 Bronx          Bronx       borough   1600.0   1600.0   
25             Bronxwood          Bronx  neighborhood      NaN      NaN   
26              Brooklyn       Brooklyn       borough   2200.0   2150.0   
27      Brooklyn Heights       Brooklyn  neighborhood   2350.0   2300.0   
28            Brookville         Queens  neighborhood      NaN      NaN   
29           Brownsville       Brooklyn  neighborhood      NaN      NaN   
..                   ...            ...           ...      ...      ...   
172         South Queens         Queens     submarket      NaN      NaN   
173  South Richmond Hill         Queens  neighborhood      NaN      NaN   
174  Springfield Gardens         Queens  neighborhood      NaN      NaN   
175           St. Albans         Queens  neighborhood      NaN      NaN   
176        Staten Island  Staten Island       borough      NaN      NaN   
177  Stuyvesant Town/PCV      Manhattan  neighborhood      NaN      NaN   
178            Sunnyside         Queens  neighborhood   1550.0   1475.0   
179          Sunset Park       Brooklyn  neighborhood   1500.0   1500.0   
180        The Rockaways         Queens     submarket      NaN      NaN   
181          Throgs Neck          Bronx  neighborhood      NaN      NaN   
182              Tremont          Bronx  neighborhood      NaN      NaN   
183              Tribeca      Manhattan  neighborhood   6500.0   6773.0   
184   University Heights          Bronx  neighborhood   1100.0   1100.0   
185      Upper East Side      Manhattan  neighborhood   2469.0   2450.0   
186      Upper West Side      Manhattan  neighborhood   2895.0   2800.0   
187               Utopia         Queens  neighborhood      NaN      NaN   
188             Van Nest          Bronx  neighborhood      NaN      NaN   
189            Wakefield          Bronx  neighborhood      NaN      NaN   
190   Washington Heights      Manhattan  neighborhood   1550.0   1545.0   
191          West Harlem      Manhattan  neighborhood   1795.0      NaN   
192         West Village      Manhattan  neighborhood   3265.0   3395.0   
193  Westchester Village          Bronx  

# Second question - what area in Manhattan had the lowest average price between 2010 and 2018?

In [15]:
ma_rent_data = rent_data[rent_data['Borough'].str.contains('Manhattan', na=False)]
ma_rent_data

areaName    Borough      areaType  2010-01  2010-02  2010-03  \
0           All Downtown  Manhattan     submarket   3200.0   3200.0   3025.0   
1            All Midtown  Manhattan     submarket   2895.0   2800.0   2800.0   
2    All Upper East Side  Manhattan     submarket   2469.0   2450.0   2400.0   
3    All Upper Manhattan  Manhattan     submarket   1825.0   1800.0   1795.0   
4    All Upper West Side  Manhattan     submarket   2895.0   2800.0   2750.0   
9      Battery Park City  Manhattan  neighborhood   3495.0   3346.0   3268.0   
35        Central Harlem  Manhattan  neighborhood   1950.0   1950.0   1893.0   
36    Central Park South  Manhattan  neighborhood   6000.0   6000.0   5950.0   
38               Chelsea  Manhattan  neighborhood   3200.0   3400.0   3245.0   
39             Chinatown  Manhattan  neighborhood   2900.0   2900.0   2550.0   
41          Civic Center  Manhattan  neighborhood      NaN      NaN      NaN   
62           East Harlem  Manhattan  neighborhood   1895.0   1900.0   1895.0   
65          East Village  Manhattan  neighborhood   2500.0   2500.0   2400.0   
69    Financial District  Manhattan  neighborhood   3198.0   3100.0   3015.0   
71              Flatiron  Manhattan  neighborhood   3895.0   3750.0   3550.0   
83         Gramercy Park  Manhattan  neighborhood   3100.0   3000.0   2995.0   
86     Greenwich Village  Manhattan  neighborhood   3200.0   3200.0   2979.0   
88      Hamilton Heights  Manhattan  neighborhood   1650.0   1700.0   1688.0   
94                Inwood  Manhattan  neighborhood   1400.0   1395.0   1395.0   
105         Little Italy  Manhattan  neighborhood   3295.0   3050.0   3000.0   
109      Lower East Side  Manhattan  neighborhood   2295.0   2250.0   2295.0   
110            Manhattan  Manhattan       borough   2800.0   2795.0   2728.0   
112       Manhattanville  Manhattan  neighborhood      NaN      NaN      NaN   
113          Marble Hill  Manhattan  neighborhood      NaN      NaN      NaN   
118              Midtown  Manhattan  neighborhood   3800.0   3900.0   3925.0   
119         Midtown East  Manhattan  neighborhood   2695.0   2625.0   2650.0   
120        Midtown South  Manhattan  neighborhood   3198.0   3200.0   3250.0   
121         Midtown West  Manhattan  neighborhood   2600.0   2650.0   2650.0   
124  Morningside Heights  Manhattan  neighborhood   2200.0   2150.0   2100.0   
131               Nolita  Manhattan  neighborhood   3248.0   2825.0   3195.0   
162     Roosevelt Island  Manhattan  neighborhood   2577.0   2555.0   2404.0   
167                 Soho  Manhattan  neighborhood   5200.0   5288.0   5275.0   
177  Stuyvesant Town/PCV  Manhattan  neighborhood      NaN      NaN      NaN   
183              Tribeca  Manhattan  neighborhood   6500.0   6773.0   6500.0   
185      Upper East Side  Manhattan  neighborhood   2469.0   2450.0   2400.0   
186      Upper West Side  Manhattan  neighborhood   2895.0   2800.0   2750.0   
190   Washington Heights  Manhattan  neighborhood   1550.0   1545.0   1597.0   
191          West Harlem  Manhattan  neighborhood   1795.0      NaN      NaN   
192         West Village  Manhattan  neighborhood   3265.0   3395.0   3113.0   

     2010-04  2010-05  2010-06  2010-07   ...     2018-05  2018-06  2018-07  \
0     3100.0   3100.0   3200.0   3195.0   ...      3825.0   3800.0   3795.0   
1     2850.0   2900.0   2950.0   3000.0   ...      3500.0   3500.0   3500.0   
2     2500.0   2550.0   2550.0   2595.0   ...      2850.0   2900.0   2925.0   
3     1800.0   1823.0   1850.0   1875.0   ...      2395.0   2400.0   2430.0   
4     2800.0   2798.0   2795.0   2800.0   ...      3300.0   3350.0   3322.0   
9     3295.0   3100.0   3395.0   3200.0   ...      4650.0   4525.0   4750.0   
35    1863.0   1900.0   1900.0   1850.0   ...      2500.0   2500.0   2550.0   
36    6500.0   6200.0   5800.0   5575.0   ...      6800.0   6850.0   7500.0   
38    3200.0   3200.0   3395.0   3400.0   ...      3995.0   4003.0   3998.0   
39    2200.0   2200.0   2

In [19]:
ma_rent_data.loc[ma_rent_data['2010-01'].idxmin()]

areaName          Inwood
Borough        Manhattan
areaType    neighborhood
2010-01             1400
2010-02             1395
2010-03             1395
2010-04             1395
2010-05             1350
2010-06             1349
2010-07             1363
2010-08             1450
2010-09             1595
2010-10             1595
2010-11             1595
2010-12             1600
2011-01             1700
2011-02             1595
2011-03             1573
2011-04             1600
2011-05             1573
2011-06             1395
2011-07             1500
2011-08             1468
2011-09             1600
2011-10             1625
2011-11             1450
2011-12             1510
2012-01             1475
2012-02             1450
2012-03             1465
                ...     
2016-09             1800
2016-10             1795
2016-11             1795
2016-12             1795
2017-01             1750
2017-02             1800
2017-03             1850
2017-04             1875
2017-05             1881


# Inwood had the cheapest apartments for rent in Manhattan.  

For my final dataset, I chose to examine the SAT scores of all New York High Schools in 2010.

In [40]:
sat_data = pd.read_csv('2010_SAT__College_Board__School_Level_Results.csv')
sat_data

DBN                                        School Name  \
0    01M292     Henry Street School for International Studies    
1    01M448               University Neighborhood High School    
2    01M450                   East Side Community High School    
3    01M458                      SATELLITE ACADEMY FORSYTH ST    
4    01M509                                  CMSP HIGH SCHOOL    
5    01M515           Lower East Side Preparatory High School    
6    01M539       New Explorations into Sci, Tech and Math HS    
7    01M650                              CASCADES HIGH SCHOOL    
8    01M696                    BARD HIGH SCHOOL EARLY COLLEGE    
9    02M047                       AMERICAN SIGN LANG ENG DUAL    
10   02M288                       FOOD AND FNANCE HIGH SCHOOL    
11   02M294                     HIGH SCHOOL FOR HIST AND COMM    
12   02M296             High School of Hospitality Management    
13   02M298                                  PACE HIGH SCHOOL    
14   02M300  Urban Assembly School of Design and Construction    
15   02M303                         The Facing History School    
16   02M305      Urban Assembly Academy of Government and Law    
17   02M308                      LOWER MANHATTAN ARTS ACADEMY    
18   02M313                          The James Baldwin School    
19   02M316  Urban Assembly School of Business for Young Wo...   
20   02M374                         GRAMERCY ARTS HIGH SCHOOL    
21   02M376                                       NYC iSchool    
22   02M400                     HIGH SCHOOL ENVRNMNTL STUDIES    
23   02M407             Institute for Collaborative Education    
24   02M408                Professional Performng Arts School    
25   02M411                 Baruch College Campus High School    
26   02M412    New York City Laboratory School Collab Studies    
27   02M413                              SCHOOL OF THE FUTURE    
28   02M414                       NEW YORK CITY MUSEUM SCHOOL    
29   02M416                     ELEANOR ROOSEVELT HIGH SCHOOL    
..      ...                                                ...   
430  30Q555                             NEWCOMERS HIGH SCHOOL    
431  30Q575                       ACADEMY OF AMERICAN STUDIES    
432  30Q580                     BACCALAUREATE SCH GLOBAL EDUC    
433  31R064       Gaynor McCown Expeditionary Learning School    
434  31R080                         MICHAEL J PETRIDES SCHOOL    
435  31R440                              NEW DORP HIGH SCHOOL    
436  31R445                         PORT RICHMOND HIGH SCHOOL    
437  31R450                                CURTIS HIGH SCHOOL    
438  31R455                           TOTTENVILLE HIGH SCHOOL    
439  31R460                        SUSAN E WAGNER HIGH SCHOOL    
440  31R470                               CONCORD HIGH SCHOOL    
441  31R600                     RALPH MCKEE VOC-TECH HIGH SCH    
442  31R605               STATEN ISLAND TECHNICAL HIGH SCHOOL    
443  32K403               ACADEMY OF ENVIRONMENTAL LEADERSHIP    
444  32K480                              Bushwick High School    
445  32K545     EBC High School for Public Service - Bushwick    
446  32K549                    BUSHWICK HS FOR SOCIAL JUSTICE    
447  32K552                         Academy of Urban Planning    
448  32K554                     ALL CITY LEADERSHIP SECONDARY    
449  32K556   Bushwick Leaders High School for Academic Excel    
450  72X470                           South Bronx High School    
451  75K371                        LILLIAN L RASHKIS HS P371K    
452  75M035                      MANHATTAN HIGH SCHOOL - P35M    
453  75Q004                                P4Q SCHOOL (@ 179)    
454  75Q256                 P256 QUEENS SCHOOL @ SAINT MARY'S    
455  75R025                        South Richmond High School    
456  75X012                      PS12X LEWIS AND CLARK SCHOOL    
457  75X754  P754 X - Jeffrey M. Rapport School for Career ...   
458  76K460                              John Ja

# What are the top 5 high schools in NYC ranked by SAT scores

In [45]:
sat_data.sort_values('Critical Reading Mean', ascending = False).head(5)

DBN                                        School Name  \
394         27Q302  Queens High School for Information, Research, ...   
49          02M535                             Park West High School    
93          05M367                         Academy for Social Action    
343  21K412/21K411               YABC AT ABRAHAM LINCOLN HIGH SCHOOL    
206         11X456               YABC AT HARRY S. TRUMAN HIGH SCHOOL    

    Number of Test Takers Critical Reading Mean Mathematics Mean Writing Mean  
394                     s                     s                s            s  
49                      s                     s                s            s  
93                      s                     s                s            s  
343                     s                     s                s            s  
206                     s                     s                s            s

As we can see from above, there are a number of schools that submitted non-numeric data.  Let's turn those 's' strings to NaN nulls, so we can properly run a sort to get our top scores.

In [44]:
good_data = sat_data.replace('s', np.nan)
good_data

DBN                                        School Name  \
0    01M292     Henry Street School for International Studies    
1    01M448               University Neighborhood High School    
2    01M450                   East Side Community High School    
3    01M458                      SATELLITE ACADEMY FORSYTH ST    
4    01M509                                  CMSP HIGH SCHOOL    
5    01M515           Lower East Side Preparatory High School    
6    01M539       New Explorations into Sci, Tech and Math HS    
7    01M650                              CASCADES HIGH SCHOOL    
8    01M696                    BARD HIGH SCHOOL EARLY COLLEGE    
9    02M047                       AMERICAN SIGN LANG ENG DUAL    
10   02M288                       FOOD AND FNANCE HIGH SCHOOL    
11   02M294                     HIGH SCHOOL FOR HIST AND COMM    
12   02M296             High School of Hospitality Management    
13   02M298                                  PACE HIGH SCHOOL    
14   02M300  Urban Assembly School of Design and Construction    
15   02M303                         The Facing History School    
16   02M305      Urban Assembly Academy of Government and Law    
17   02M308                      LOWER MANHATTAN ARTS ACADEMY    
18   02M313                          The James Baldwin School    
19   02M316  Urban Assembly School of Business for Young Wo...   
20   02M374                         GRAMERCY ARTS HIGH SCHOOL    
21   02M376                                       NYC iSchool    
22   02M400                     HIGH SCHOOL ENVRNMNTL STUDIES    
23   02M407             Institute for Collaborative Education    
24   02M408                Professional Performng Arts School    
25   02M411                 Baruch College Campus High School    
26   02M412    New York City Laboratory School Collab Studies    
27   02M413                              SCHOOL OF THE FUTURE    
28   02M414                       NEW YORK CITY MUSEUM SCHOOL    
29   02M416                     ELEANOR ROOSEVELT HIGH SCHOOL    
..      ...                                                ...   
430  30Q555                             NEWCOMERS HIGH SCHOOL    
431  30Q575                       ACADEMY OF AMERICAN STUDIES    
432  30Q580                     BACCALAUREATE SCH GLOBAL EDUC    
433  31R064       Gaynor McCown Expeditionary Learning School    
434  31R080                         MICHAEL J PETRIDES SCHOOL    
435  31R440                              NEW DORP HIGH SCHOOL    
436  31R445                         PORT RICHMOND HIGH SCHOOL    
437  31R450                                CURTIS HIGH SCHOOL    
438  31R455                           TOTTENVILLE HIGH SCHOOL    
439  31R460                        SUSAN E WAGNER HIGH SCHOOL    
440  31R470                               CONCORD HIGH SCHOOL    
441  31R600                     RALPH MCKEE VOC-TECH HIGH SCH    
442  31R605               STATEN ISLAND TECHNICAL HIGH SCHOOL    
443  32K403               ACADEMY OF ENVIRONMENTAL LEADERSHIP    
444  32K480                              Bushwick High School    
445  32K545     EBC High School for Public Service - Bushwick    
446  32K549                    BUSHWICK HS FOR SOCIAL JUSTICE    
447  32K552                         Academy of Urban Planning    
448  32K554                     ALL CITY LEADERSHIP SECONDARY    
449  32K556   Bushwick Leaders High School for Academic Excel    
450  72X470                           South Bronx High School    
451  75K371                        LILLIAN L RASHKIS HS P371K    
452  75M035                      MANHATTAN HIGH SCHOOL - P35M    
453  75Q004                                P4Q SCHOOL (@ 179)    
454  75Q256                 P256 QUEENS SCHOOL @ SAINT MARY'S    
455  75R025                        South Richmond High School    
456  75X012                      PS12X LEWIS AND CLARK SCHOOL    
457  75X754  P754 X - Jeffrey M. Rapport School for Career ...   
458  76K460                              John Ja

Once more, let's do the sort and we'll have our top scores.

In [46]:
good_data.sort_values('Critical Reading Mean', ascending = False).head(5)

DBN                                     School Name  \
42   02M475                         STUYVESANT HIGH SCHOOL    
442  31R605            STATEN ISLAND TECHNICAL HIGH SCHOOL    
384  25Q525  Townsend Harris High School at Queens College    
193  10X696       HS of American Studies at Lehman College    
185  10X445                   BRONX HIGH SCHOOL OF SCIENCE    

    Number of Test Takers Critical Reading Mean Mathematics Mean Writing Mean  
42                    804                   674              735          678  
442                   287                   638              673          617  
384                   273                   637              644          642  
193                    74                   635              630          619  
185                   683                   632              685          643

# These were the top 5 schools sorted by their SAT critical reading mean scores.  